### PR TITLE
Support string tensor for tensor context

### LINKED ISSUE #1399: [BUG] App `bfs_edges` raise error if the `depth_limit` is `None`
**Describe the bug**

```python
home/xxxxpython3.7/lib/python3.7/site-packages/graphscope/framework/utils.py in pack(v)
    355         param.Pack(data_types_pb2.BytesValue(value=v))
    356     else:
--> 357         raise ValueError("Wrong type of query param {}".format(type(v)))
    358     return param
    359 

ValueError: Wrong type of query param <class 'NoneType'>
```

**To Reproduce**
```python
sess.nx().builtin.bfs_edges(G,source=1)
```

**Additional context**
- [ ] find a way to  support pass `depth_limit` as `None`
- [ ] check other builtin apps if have the same issue.
- [ ] make sure builtin document is consistent with app behavior.

### PR DESCRIPTION
**Pull request by acezen**
## Related issue number

Fixes #1399

### CODE REVIEW

File: analytical_engine/core/utils/trivial_tensor.h [hunk 1]
```diff
@@ -69,5 +70,40 @@ struct trivial_tensor_t {
...
+ public:
+  trivial_tensor_t() : size_(0) {}
+
+  ~trivial_tensor_t() = default;
+
+  std::vector<std::string>& data() { return data_; }
+
+  const std::vector<std::string>& data() const { return data_; }
+
+  void fill(const std::string& value) { data_.push_back(value); }
```
Comment (sighingnow) [inline analytical_engine/core/utils/trivial_tensor.h:85]: `fill` usually means **filling all cells with the given value** (see https://en.cppreference.com/w/cpp/algorithm/fill).

I think here you are meaning something like `append` or `push_back`.

File: analytical_engine/core/utils/trivial_tensor.h [hunk 2]
```diff
@@ -69,5 +70,40 @@ struct trivial_tensor_t {
...
+    if (flat_size != size_) {
+      data_.resize(flat_size);
+      size_ = flat_size;
+    }
+  }
+
+ private:
+  size_t size_;
+  std::vector<size_t> shape_;
+  std::vector<std::string> data_;
```
Comment (sighingnow) [inline analytical_engine/core/utils/trivial_tensor.h:106]: If you only need to "Append" values, I think `arrow::StringArray` would be more suitable and more efficient.

File: analytical_engine/core/context/tensor_context.h [hunk 3]
```diff
@@ -473,6 +525,244 @@ class TensorContextWrapper : public ITensorContextWrapper {
...
+      }
+      gather_archives(*arc, comm_spec, old_size);
+    }
+    return arc;
+  }
+
+  bl::result<vineyard::ObjectID> ToVineyardTensor(
+      const grape::CommSpec& comm_spec, vineyard::Client& client,
+      uint32_t axis) override {
+    return 0;
```
Comment (sighingnow) [inline analytical_engine/core/context/tensor_context.h:645]: return `bl::new_error("not supported")` or something like that.

File: analytical_engine/core/utils/trivial_tensor.h [hunk 4]
```diff
@@ -69,5 +72,40 @@ struct trivial_tensor_t {
...
+ public:
+  trivial_tensor_t() : size_(0), data_(nullptr) {}
+
+  ~trivial_tensor_t() = default;
+
+  std::shared_ptr<arrow::StringArray>& data() { return data_; }
+
+  const std::shared_ptr<arrow::StringArray>& data() const { return data_; }
+
+  void fill(const std::string& value) { return; }
```
Comment (sighingnow) [inline analytical_engine/core/utils/trivial_tensor.h:87]: `LOG(ERROR)` some message in unsupported/unimplemented methods.

**Review by sighingnow — APPROVED** [review]

File: analytical_engine/core/utils/trivial_tensor.h (hunk 1 above)
Comment (acezen) [reply analytical_engine/core/utils/trivial_tensor.h]: fix

File: analytical_engine/core/utils/trivial_tensor.h (hunk 2 above)
Comment (acezen) [reply analytical_engine/core/utils/trivial_tensor.h]: Updated

**Pull request merged**